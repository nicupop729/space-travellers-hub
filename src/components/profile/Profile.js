import { useSelector } from 'react-redux';
import Component from './Component';
import './profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  const missions = useSelector((state) => state.missionsReducer);
  const activeRockets = rockets.filter((rocket) => rocket.reserved === true);
  const activeMissions = missions.filter(
    (mission) => mission.reserved === true,
  );

  return (
    <div className="Reservations">
      <Component
        reservations={activeMissions}
        object="Missions"
        message="No missions joined"
        type="mission"
      />
      <Component
        reservations={activeRockets}
        object="Rockets"
        message="No rockets reserved"
        type="rocket"
      />
    </div>
  );
};

export default Profile;
