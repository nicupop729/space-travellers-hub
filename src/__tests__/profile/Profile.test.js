import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Profile from '../../components/profile/Profile';

describe('snapshot testing', () => {
  test('snapshot for Profile component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
