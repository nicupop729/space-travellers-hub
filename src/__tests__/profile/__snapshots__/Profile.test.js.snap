// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Profile component 1`] = `
<div
  className="Reservations"
>
  <div
    className="Reservations__div"
  >
    <h1>
      My
      Missions
    </h1>
    <p
      className="Reservations__div_none"
    >
      No missions joined
    </p>
  </div>
  <div
    className="Reservations__div"
  >
    <h1>
      My
      Rockets
    </h1>
    <p
      className="Reservations__div_none"
    >
      No rockets reserved
    </p>
  </div>
</div>
`;
